central/resourcecollection/service/service_impl.go

```diff
@@ -155,6 +157,17 @@ func (s *serviceImpl) DeleteCollection(ctx context.Context, request *v1.Resource
 	if request.GetId() == "" {
 		return nil, errors.Wrap(errox.InvalidArgs, "Non empty collection id must be specified to delete a collection")
 	}
+	// error out if collection is in use by a report config
+	err := s.reportConfigDatastore.Walk(ctx, func(reportConfig *storage.ReportConfiguration) error {
+		if reportConfig.GetScopeId() == request.GetId() {
+			return errors.New("Collection is in use in one or more report configurations")
+		}
+		return nil
```

Comment:
Why don't we just add a search tag to the field in the proto and query to see if any results come up when we search for report configurations with a particular id relating to the collection in question?

Reply:
The other thing here is once we migrate to collections and postgres by default, we could define the scopeid as a foreign key in the proto and then the postgres datastore will handle all of the checks for us

Reply:
Thought about adding a search tag to the report_configuratons proto, but it would change the schema. So @md2119 suggested that it would be better to just walk through the report configs and check for now. It shouldn't take too long either since there won't be too many reportConfigs in the db anyways.

Reply:
We can remove the check once scope_id becomes a foreign key.

Reply:
Yes, we have frozen the postgres schema for 3.74. Adding fk to report configuration will change its schema. Although there is a path through the weeds to make it happen, we need to keep the risk minimum for a blotch-free postgres release.

Reply:
Understood, thanks for the clarity